src/system/AdminModule/Controller/AdminInterfaceController.php

```diff
@@ -0,0 +1,434 @@
+                    'security' => $data
+        ]);
+    }
+
+    /**
+     * @Route("/updatecheck")
+     *
+     * Add update check
+     *
+     * @return Response symfony response object
+     */
+    public function updatecheckAction() {
+        if (!$this->get('zikula_extensions_module.api.variable')->get('ZConfig', 'updatecheck')) {
+            return [
+                'update_show' => false
+            ];
+        }
+        $force = false;
+
+        $now = time();
+        $lastChecked = (int) $this->get('zikula_extensions_module.api.variable')->get('ZConfig', 'updatelastchecked');
+        $checkInterval = (int) $this->get('zikula_extensions_module.api.variable')->get('ZConfig', 'updatefrequency') * 86400;
+        $updateversion = $this->get('zikula_extensions_module.api.variable')->get('ZConfig', 'updateversion');
+        $update_show = false;
+        $update_version = false;
+
+        if ($force == false && (($now - $lastChecked) < $checkInterval)) {
+            // dont get an update because TTL not expired yet
+            $onlineVersion = $updateversion;
+        } else {
+            $this->get('zikula_extensions_module.api.variable')->set('ZConfig', 'updatelastchecked', (int) time());
+
+            $onlineVersion = '';
+            $newVersionInfo = trim($this->zcurl('https://api.github.com/repos/zikula/core/releases'));
+            if ($newVersionInfo === '') {
+                $update_show = false;
+            }
+            $newVersionInfo = json_decode($newVersionInfo, true);
+            if (!is_array($newVersionInfo) || isset($newVersionInfo['message']) /* Will be set if rate limits encountered */) {
+                $update_show = false;
+            }
+
+            foreach ($newVersionInfo as $version) {
+                if (!is_array($version)) {
+                    // Invalid response, probably api limits encountered.
+                    $update_show = false;
+                }
+                if (!array_key_exists('prerelease', $version) || $version['prerelease']) {
+                    continue;
+                }
+                if (array_key_exists('tag_name', $version)) {
+                    if (version_compare($version['tag_name'], $onlineVersion) == 1) {
+                        $onlineVersion = $version['tag_name'];
+                    }
+                }
+            }
+
+            if ($onlineVersion == '') {
+                $update_show = false;
+            }
+            $this->get('zikula_extensions_module.api.variable')->set('ZConfig', 'updateversion', $onlineVersion);
+        }
+
+        // compare with db Version_Num
+        // if 1 then there is a later version available
+        if (version_compare($onlineVersion, $this->get('zikula_extensions_module.api.variable')->get('ZConfig', 'Version_Num') == 1)) {
+            $update_show = true;
+            $update_version = $onlineVersion;
+        } else {
+            $update_show = false;
```

Comment:
this (and line 247) overwrites all previous values of `$update_show` so why is it set previously?